Feature/gpu

leoTiez opened the pull request:
Class implementation to use the network on both gpu and cpu.

leoTiez committed:
Update: Started class implementation. Not tested tho

leoTiez committed:
Update: finished class implementation

Network architecture and training phase implementation
are moved to a class implementation for easy reuse

Commit:
WIP

leoTiez committed:
Update: Requires tensorflow gpu now

leoTiez committed:
Update: Investigating problem with feeding in data when running tests

leoTiez committed:
Update: Changed while loop to test whether this causes dead ends

leoTiez committed:
Update: Removed while loop and cleaned code

Since the cuda implementation has difficulties analysing
the tesnforflow while loop properly, it was replaced by
the previous for loop implementation. After checking
the tensorflow graph it is assumed that the gradient computation
and the forward pass work similarly and give the same results.

Moreover, the code has been cleaned up-

leoTiez committed:
Bigfix: Fixed too many values to unpack and renamed variables

leoTiez committed:
Bigfix: Changed valdidation to test set

leoTiez committed:
Update: Changed it to class implementation

leoTiez committed:
Update: added .save files

leoTiez committed:
Update: Set batch size as placeholder

leoTiez committed:
Merge branch 'feature/classImplementation' into feature/gpu

# Conflicts:
#	RCNN_tf.py

leoTiez committed:
Update: Added possibility of setting a device

Makes it possible to switch between cpu and gpu implementation

leoTiez committed:
Bugfix: Changed variable name

leoTiez committed:
Update: Added raise error if lrn depth is not properly set

When running in gpu mode the depth for the lrn operation
(normalization_feature_maps) must not be greater than 7
and smaller than 1. Error message was added

leoTiez committed:
Bugfix: Changed upper bound for rln depth value check

leoTiez committed:
Update: Updated readme accordong to the recent class implementation

leoTiez committed:
Update: Added both tensorflow and tensorflow-gpu

leoTiez committed:
Bugfix: Removed duplicated keyword in dict

leoTiez committed:
Test: Used tensorflow accuracy metric to test whether this works on gpu

leoTiez committed:
Update: Changed accuracy since tf implementation does not work on gpu

Also set parameters explicitly in main file

bt-s committed:
Create a flag for RCNN.train for printing information about vars

If the flag is set to True, the names of the trainable variables,
plus their shapes, as well as the total number of trainable variables
in the network will be printed.

bt-s committed:
Merge pull request #22 from leoTiez/feature/print-var-info

Feature/print var info

adrianc-a commented on RCNN_tf.py:
```diff
@@ -157,14 +193,266 @@ def accuracy(
+        with tf.Session() as sess:
+            if create_graph:
+                writer = tf.summary.FileWriter('logs/.')
+                writer.flush()
```
do we need to this? it seems unnecessary (correct me if i'm wrong). if we need it i feel like it would be after we call `add_graph` not when we just create the writer, besides I would expect `close()` to flush anyways.

leoTiez replied:
I found this post on stackoverflow.
https://stackoverflow.com/questions/52483296/when-do-i-have-to-use-tensorflows-filewriter-flush-method

So it doesn't harm us, but we can indeed remove it

bt-s committed:
Update .gitignore

bt-s committed:
WIP

bt-s committed:
Update gitignore for VIM

bt-s committed:
Create generic load_batch file

bt-s committed:
Write file introduction for RCNN_tf.py

bt-s committed:
Suppress warnings

bt-s committed:
Rename some variables and reformat code for improved readibility

bt-s committed:
Create flag for printing information about trainable parameters

bt-s committed:
WIP improve main function

bt-s committed:
Fix merge conflicts

bt-s committed:
Remove obsolete code from load_data.py

bt-s committed:
Provide docstrings following Google spec

leoTiez committed:
Udate: Removed unnecessary library imports

leoTiez committed:
Update: added missing docstrings

Also. tuple to list for stride parameter to be consistent.

While loop in cpu execution has been slightly adapted
since empirical tests revealed it works better.
Todo comment left for investigation

leoTiez committed:
Merge pull request #25 from leoTiez/bugfix/accuracy

Bugfix/accuracy

leoTiez committed:
Merge pull request #23 from leoTiez/feature/main-files

Feature/main files

adrianc-a committed:
Splitting main and RCNN

leoTiez committed:
Merge pull request #26 from leoTiez/feature/main-files

Splitting main and RCNN

bt-s committed:
Use He initialization

bt-s committed:
Use AdagradOptimizer

bt-s committed:
Change the parameters for the model training on MNIST

leoTiez committed:
Merge pull request #27 from leoTiez/feature/he-initialization

Feature/he initialization

leoTiez committed:
Update: Changed parameters and added greyscale flag

leoTiez committed:
Update: Added greyscale flag for cifar data sets

leoTiez committed:
Update: Investigation of the recurrent layer

Layer is slightly worse than without any recurrent input.
Probably problem with bias term.

leoTiez committed:
Bugfix: Removed return of subset

leoTiez committed:
Update: Removed unnecessary loop for gpu

The loop could be removed since a flag solved the issue of dead
ends

leoTiez committed:
Update: Changed different spelling of gray to be consistent

leoTiez committed:
Update: Tests and tweaking of parameters

leoTiez committed:
'Update: Added parameters to determine while loop behavior

leoTiez committed:
Update: Changed the depeth for cifar10 back to 3

leoTiez committed:
Update: Changed the optimiser to gradient descent

leoTiez committed:
Update: parameter tweaking, increasing epochs and learning rate

leoTiez committed:
Bugfix: Compute cell states after first output computation

The cell states are calculated with the feedforward
output first

leoTiez committed:
Update: Change of hyperparameters for better learning performance

leoTiez committed:
Update: Clean up

Removed unnecessary parameters and comments

leoTiez committed:
Update: Implemented adaptive learnign rate

Learning rate is  now dynamically passed
When the accuracy does not increase for a number of epochs,
it is decreased by multiplying by a constant factor.
If adpative learning factor is set 1, no adaptive learning
is performed.

Updated docstrings

leoTiez committed:
Update: Changed default parameter for adaptive learning factor

leoTiez committed:
Update: Changed hyperparameter of mnist according to changes in cifar

leoTiez committed:
Update: Passing dropout rate as a parameter

Instead of setting it statically to 0.2, the dropout
rate can be passed as a parameter to the function train

leoTiez committed:
Merge pull request #29 from leoTiez/feature/adaptiveLearningRate

Feature/adaptive learning rate

leoTiez committed:
Update: Replaced re initialization of global function

The function for the initialization has been set to a global
variable. Thus, re initialization is not necessary.
The concerning lines have been removed.

Spelling has been changed to capital letters

leoTiez committed:
Merge pull request #28 from leoTiez/bugfix/accuracyTests

Bugfix/accuracy tests

leoTiez committed:
Update: Removed unnecessary flush statement for writer object
